bot/cogs/filtering.py

```diff
@@ -250,11 +262,12 @@ def mod_log(self) -> ModLog:
         This will match the expression even if it does not have boundaries before and after.
         """
         for regex_pattern in TOKEN_WATCHLIST_PATTERNS:
-            if regex_pattern.search(text):
+            match = regex_pattern.search(text)
+            if match:
 
                 # Make sure it's not a URL
                 if not URL_RE.search(text):
-                    return True
+                    return match  # match objects always have a boolean value of True
```

Comment:
It might resolve to bool(match) == True, but it's not returning that so the type annotation is incorrect.